scripts/core/ui/components/content-create-dropdown/more-templates.tsx

```diff
@@ -28,18 +29,34 @@ export class MoreTemplates extends React.PureComponent<IProps, IState> {
     }
 
     fetchData(pageToFetch: number, pageSize: number, abortSignal?: AbortSignal): Promise<IRestApiResponse<ITemplate>> {
+        const templateDesks = nameof<ITemplate>('template_desks');
+        const currentDeskId = sdApi.desks.getCurrentDeskId();
+
+        const deskCriteria: any = [
+            {[templateDesks]: {$exists: false}},
+            {[templateDesks]: []},
+        ];
+
+        if (currentDeskId != null) {
+            deskCriteria.push({[templateDesks]: {$in: [currentDeskId]}});
+        }
+
+        const criteria = {$or: [{$or: deskCriteria}, {user: sdApi.user.getCurrentUserId()}]};
+        const templateName = nameof<ITemplate>('template_name');
+
         return httpRequestJsonLocal<IRestApiResponse<ITemplate>>({
             method: 'GET',
             path: '/content_templates',
             urlParams: {
                 max_results: pageSize,
                 page: pageToFetch,
-                sort: nameof<ITemplate>('template_name'),
-                where: this.state.searchString.length < 1 ? undefined : {
-                    [nameof<ITemplate>('template_name')]: {
+                sort: templateName,
+                where: this.state.searchString.length < 1 ? {$and: [criteria]} : {
```

Comment:
```
const where = {$and: [criteria]};

if (this.state.searchString.length < 1) {
    where[templateName] = {
        $regex: this.state.searchString,
        $options: '-i',
    },
}

httpRequestJsonLocal(where, ...)
```

Avoid repeating the same logic twice ( `{$and: [criteria]}` ). In my pseudo-code it's explicit that `{$and: [criteria]}` is always applied and search is applied conditionally. In your version it wouldn't be that clear for someone adding more filters. Should he add it on both places or only one for example? It is clear at the moment, but as code grows it becomes an issue.